server/src/main/java/org/elasticsearch/env/NodeEnvironment.java

```diff
@@ -1044,15 +1044,8 @@ public String nodeId() {
         final Set<ShardId> shardIds = new HashSet<>();
         final String indexUniquePathId = index.getUUID();
         for (final NodePath nodePath : nodePaths) {
-            Path location = nodePath.indicesPath;
-            if (Files.isDirectory(location)) {
-                try (DirectoryStream<Path> indexStream = Files.newDirectoryStream(location)) {
-                    for (Path indexPath : indexStream) {
-                        if (indexUniquePathId.equals(indexPath.getFileName().toString())) {
-                            shardIds.addAll(findAllShardsForIndex(indexPath, index));
-                        }
-                    }
-                }
+            if (Files.isDirectory(nodePath.indicesPath)) {
```

Comment:
Shouldn't this be checking `nodePath.indicesPath.resolve(indexUniquePathId)`?

Reply:
This preserves the check in the original code; we already check `Files.isDirectory(nodePath.indicesPath.resolve(indexUniquePathId))` within `findAllShardsForIndex`. Maybe we should just remove this check entirely?